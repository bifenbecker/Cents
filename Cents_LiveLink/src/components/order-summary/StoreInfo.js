import React from "react";
import {Flex, Text} from "rebass/styled-components";

import {FullScreenModalForm} from "../common";

const StoreInfo = props => {
  const {
    orderDetails: {
      store: {
        name,
        address,
        city,
        state,
        zipCode,
        phoneNumber,
        dcaLicense,
        commercialDcaLicense,
      },
    },
    onClose,
  } = props;

  return (
    <FullScreenModalForm
      header="Store Info"
      onClose={onClose}
      onSubmit={onClose}
      btnLabel="Close"
    >
      <Flex
        width="100%"
        flexDirection="column"
        justifyContent={["flex-start", "flex-start", "flex-start", "center"]}
        alignItems="center"
        textAlign="center"
        pb={"10px"}
      >
        <Text>{name}</Text>
        <Text>
          {address}, {city}
        </Text>
        <Text>
          {state}, {zipCode}
        </Text>
        <Text>{phoneNumber}</Text>
      </Flex>
      {dcaLicense && (
        <Flex {...styles.dcaContainer}>
          <Text {...styles.dcaHeader}>DCA # - Retail:</Text>
          <Text {...styles.dcaInfo}>{dcaLicense}</Text>
        </Flex>
      )}
      {commercialDcaLicense && (
        <Flex {...styles.dcaContainer}>
          <Text {...styles.dcaHeader}>DCA # - Commercial:</Text>
          <Text {...styles.dcaInfo}>{commercialDcaLicense}</Text>
        </Flex>
      )}
    </FullScreenModalForm>
  );
};

const styles = {
  dcaContainer: {
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    py: "6px",
  },
  dcaHeader: {
    fontFamily: "secondary",
    pt: "6px",
    pb: "4px",
    lineHeight: 1.5,
    flexDirection: "column",
  },
  dcaInfo: {
    fontFamily: "secondary",
    py: "2px",
    lineHeight: 1.5,
    flexDirection: "column",
  },
};

export default StoreInfo;
