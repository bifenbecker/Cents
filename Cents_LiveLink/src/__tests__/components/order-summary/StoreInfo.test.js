import React from "react";
import {render, fireEvent} from "@testing-library/react";

import StoreInfo from "../../../components/order-summary/StoreInfo";

const onClose = jest.fn();
const renderStoreInfo = props => {
  const {container, getByText, getByAltText} = render(
    <StoreInfo {...props} onClose={onClose} />
  );

  return {container, getByText, getByAltText};
};

describe("StoreInfo component", () => {
  it("Should match snapshot", () => {
    const {container} = renderStoreInfo({
      orderDetails: {
        store: {
          name: "Store one",
          address: "Address of the store one",
          city: "Pune",
          state: "Maharashtra",
          zipCode: "413523",
          phoneNumber: "54765486458",
          dcaLicense: true,
          commercialDcaLicense: true,
        },
      },
    });

    expect(container).toMatchSnapshot();
  });

  it("Should close modal when clicked on cross icon", () => {
    const {getByAltText} = renderStoreInfo({
      orderDetails: {
        store: {
          name: "Store one",
          address: "Address of the store one",
          city: "Pune",
          state: "Maharashtra",
          zipCode: "413523",
          phoneNumber: "54765486458",
          dcaLicense: true,
          commercialDcaLicense: true,
        },
      },
    });
    const closeModal = getByAltText("Modal Close");
    fireEvent.click(closeModal);

    expect(onClose).toHaveBeenCalled();
  });
});
