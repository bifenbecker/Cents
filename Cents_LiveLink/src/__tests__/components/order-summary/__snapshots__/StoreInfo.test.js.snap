// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreInfo component Should match snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ sc-gsnTZi dWNzKB hzPtvw"
    height="var(--app-height)"
  >
    <img
      alt="Modal Close"
      class="sc-bczRLJ gKoBJV"
      height="20px"
      src="Icon_Mobile_Exit_Flow.svg"
      width="20px"
    />
    <div
      class="sc-bczRLJ sc-gsnTZi AXwHa hzPtvw"
      height="67px"
    >
      <div
        class="sc-bczRLJ jCICIV"
        font-size="1.25rem,1.5rem"
      >
        Store Info
      </div>
    </div>
    <div
      class="sc-bczRLJ dwQaiA"
      height="calc(var(--app-height) - 67px - 6rem)"
    >
      <div
        class="sc-bczRLJ hMLfYu"
        height="calc(var(--app-height) - 67px - 6rem)"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi fIlruQ hzPtvw"
          width="100%"
        >
          <div
            class="sc-bczRLJ jITePI"
          >
            Store one
          </div>
          <div
            class="sc-bczRLJ jITePI"
          >
            Address of the store one
            , 
            Pune
          </div>
          <div
            class="sc-bczRLJ jITePI"
          >
            Maharashtra
            , 
            413523
          </div>
          <div
            class="sc-bczRLJ jITePI"
          >
            54765486458
          </div>
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi elTHjs hzPtvw"
        >
          <div
            class="sc-bczRLJ bXdFtq"
            font-family="secondary"
          >
            DCA # - Retail:
          </div>
          <div
            class="sc-bczRLJ hgNSXQ"
            font-family="secondary"
          />
        </div>
        <div
          class="sc-bczRLJ sc-gsnTZi elTHjs hzPtvw"
        >
          <div
            class="sc-bczRLJ bXdFtq"
            font-family="secondary"
          >
            DCA # - Commercial:
          </div>
          <div
            class="sc-bczRLJ hgNSXQ"
            font-family="secondary"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-bczRLJ sc-gsnTZi hNbvdg hzPtvw"
      height="4rem"
    >
      <button
        class="sc-bczRLJ eSiIwW"
        font-size="3,4"
        height="3.5rem,4rem"
        width="100%,100%,100%,50%"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
